pkgs/v2/default.nix

```diff
@@ -0,0 +1,252 @@
+      };
+    });
+
+  # does this attrset have an enable option? ie is this a module?
+  hasEnableOption = options: hasAttr "enable" options && options.enable._type == "option";
+
+  # given an attrset of nested options, return a list of
+  # NixModulesV2RegistryEntry objects in the Replit protobuf protocol
+  getModulesFromOptions = options: path:
+    foldl' (acc: attrName:
+      let
+        value = options.${attrName};
+      in
+        if hasEnableOption value
+        then
+          let
+            enable = value.enable;
+            module = {
+              id = concatStringsSep "." path;
+              name = enable.moduleName;
+              description = enable.moduleDescription;
+              options = getModuleOptions value;
+            };
+          in acc ++ [module]
+        else
+          acc ++ (getModulesFromOptions value (path ++ [attrName]))
+    ) [] (attrNames options);
+
+  # given an option, return an attrset containing the type field to use for the
+  # NixModuleOption in the Replit protobuf protocol
+  getTypeFieldForOption = option:
+    let type = option.type.functor.name;
+    in if type == "bool"
+      then
+        {
+          booleanType = {
+            default = option.default;
+          };
+        }
+      else if type == "enum"
+      then
+        {
+          choiceStringType = {
+            default = option.default;
+            choices = option.definitions;
+          };
+        }
+      else if type == "listOf"
+      then
+        {
+          stringListType = {
+            default = option.default;
+          };
+        }
+      else if type == "str"
+      then
+        {
+          stringType = {
+            default = option.default;
+          };
+        }
+      else if type == "int"
+      then
+        {
+          integerType = {
+            default = option.default;
+          };
+        }
+      else {};
+
+  # convert an attrset of options to a list of NixModuleOptions in the
+  # Replit protobuf protocol
+  getModuleOptions = options:
+    # Filter out internal attributes prefixed with _
+    let optionNames = filter (name: !(strings.hasPrefix "_" name)) (attrNames options);
+    in
+    foldl' (acc: name:
+      let
+        option = options.${name};
+        description = option.description;
+        type = option.type.functor.name;
+        typeField = getTypeFieldForOption option;
+      in acc ++ [({
+        inherit name description;
+      } // typeField)]
+    ) [] optionNames;
+
+  # given an option, return an attrset containing the type-specific
+  # `value` field for NixModuleConfigValue in the Replit protobuf protocol
+  getTypeSpecificValue = option:
+    let type = option.type.functor.name;
+    in
+      if type == "bool"
+      then
+        {
+          booleanValue = option.value;
+        }
+      else if type == "enum" || type == "str"
+      then
+        {
+          stringValue = option.value;
+        }
+      else if type == "listOf"
```

Comment:
Is this going to blow up if someone uses `types.listOf types.int`? Probably okay for now but maybe we should flag with a comment.

Reply:
Ok I can make it fail.

Reply:
Actually skip emit if it list of something else.